bob-backend/src/pearl/group.rs

```diff
@@ -302,8 +312,9 @@ impl Group {
         result
     }
 
-    pub async fn exist(&self, keys: &[BobKey]) -> Vec<bool> {
+    pub async fn exist(&self, keys: &[BobKey]) -> Result<Vec<bool>, Error> {
         let mut exist = vec![false; keys.len()];
+        let _reinit_lock = self.reinit_lock.try_read().map_err(|_| Error::holder_temporary_unavailable())?;
```

Comment:
Please, move this line to the very beginning of this function. In this way it will be unified with the other functions. Also `try_read` is non-blocking, so there is no point in moving any operation to a position before it